### ISSUE TITLE
Add examples for `with_tempfile()`

### ISSUE THREAD

**Issue by jimhester** [r-lib/withr#106]
Maybe showing writing to a temporary file, reading from it then showing it doesn't exist afterwards?

Wild caught usage examples in the [GitHub search results for 'with_tempfile'](https://github.com/search?q=with_tempfile+user%3Acran&type=Code).

**Comment by moodymudskipper** [comment]
Here are a few simple examples that might be real use cases: 

```
# check how `write.csv()` saves files
iris_csv <- with_tempfile("tf", {write.csv(iris, tf); readLines(tf)})
cat(iris_csv[1:2], sep="\n")
#> "","Sepal.Length","Sepal.Width","Petal.Length","Petal.Width","Species"
#> "1",5.1,3.5,1.4,0.2,"setosa"

# check how big would iris weigh if being written as csv vs RDS
with_tempfile("tf", {write.csv(iris, tf); file.size(tf)})
#> [1] 4972
with_tempfile("tf", {saveRDS(iris, tf); file.size(tf)})
#> [1] 1081

# get the command history
command_history <- with_tempfile("tf", {savehistory(tf); readLines(tf)})
tail(command_history, 3)
#> [1] "with_tempfile(\"tf\", {write.csv(iris, tf); file.size(tf)})"             
#> [2] "with_tempfile(\"tf\", {saveRDS(iris, tf); file.size(tf)})"               
#> [3] "saved_history <- with_tempfile(\"tf\", {savehistory(tf); readLines(tf)})"
```

**Comment by jimhester** [comment]
Fixed by 670e5cc
